src/tools/illink/src/linker/Linker/InterfaceImplementor.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Mono.Cecil;
+
+namespace Mono.Linker
+{
+	public class InterfaceImplementor
+	{
+		/// <summary>
+		/// The type that implements <see cref="InterfaceImplementor.InterfaceType"/>.
+		/// </summary>
+		public TypeDefinition Implementor { get; }
+		/// <summary>
+		/// The .interfaceimpl on <see cref="InterfaceImplementor.Implementor"/>that points to <see cref="InterfaceImplementor.InterfaceType"/>
+		/// </summary>
+		public InterfaceImplementation InterfaceImplementation { get; }
+		/// <summary>
+		/// The type of the interface that is implemented by <see cref="InterfaceImplementor.Implementor"/>
+		/// </summary>
+		public TypeDefinition InterfaceType { get; }
```

Comment:
`InterfaceImplementation.InterfaceType` is a `TypeReference` and I thought it would be helpful to keep the `TypeDefinition` here. We could remove it and resolve it each time we need the definition, if you think that makes more sense.

Reply:
Yeah, that's a good idea. I'll assert in the constructor.